Planck keymap

ReFil opened the pull request:
This is a port of the QMK planck keymap. Given the one for the CRBN is identical I fixed it at the same time. I think I followed the indentation guidelines this time

Nicell commented:
Would be nice to get a proper default mapping for the Planck. Although, do we have anyone with the Planck that can test this? I'm not aware of anyone at least. At the very least, I assume you've tested this mapping on your CRBN @ReFil?

ReFil commented:
Yeah it's been thoroughly tested on the CRBN, by our customers too, I haven't got a planck to test it on but it builds fine

innovaker commented:
@petejohanson was the original author of the planck shield so it might be wise to add him as a reviewer once you're satisfied with everything else @Nicell.

ReFil commented:
Hiya, just wondering what's happening with this PR as I've got some more stuff for the CRBN waiting in the wings but i wanted to get this merged first

Nicell commented on app/boards/arm/planck/planck_rev6.keymap:
```diff
@@ -1,21 +1,45 @@
+				&kp LSHFT &kp Z    &kp X &kp C &kp V &kp B  &kp N   &kp M  &kp COMMA &kp DOT &kp SLASH &kp RET
+				&trans   &kp LCTL &kp LALT &kp LGUI &mo 1 &trans &kp SPACE &mo 2 &kp LEFT &kp DOWN &kp UP &kp RIGHT
+			>;
+	 };
```
```suggestion
		};
```
Looks like just this one line was misaligned

Nicell commented on app/boards/arm/planck/planck_rev6.keymap:
```diff
@@ -1,21 +1,45 @@
 / {
 	keymap {
-		compatible = "zmk,keymap";
+	compatible = "zmk,keymap";
```
```suggestion
		compatible = "zmk,keymap";
```
I think this indentation should be back to how it was before

Nicell commented on app/boards/shields/crbn/crbn.keymap:
```diff
@@ -10,46 +10,47 @@
 / {
 	keymap {
-		compatible = "zmk,keymap";
+	compatible = "zmk,keymap";
```
```suggestion
		compatible = "zmk,keymap";
```

Nicell commented on app/boards/shields/crbn/crbn.keymap:
```diff
@@ -10,46 +10,47 @@
+				&trans   &kp LCTL &kp LALT &kp LGUI &mo 1 &kp SPACE &trans &mo 2 &kp LEFT &kp DOWN &kp UP &kp RIGHT
 			>;
-		};
+	 };
```
```suggestion
		};
```

Nicell reviewed (changes requested):
I have some indentation comments, but otherwise this looks fine. There's also a conflict now that the CRBN encoder stuff is in.

ReFil commented:
Ok I'll fix the encoder stuff tonight then :)

ReFil commented:
Encoder stuff added to crbn keymap

ReFil commented:
Fixed the conflict, changed the mapping of the encoder slightly, it does pgup and pgdn on raise layer and vol on default to try and please all our customers as we've been giving them builds with the encoder mapped to vol to date (we were waiting for this to get merged before submitting a pr for our encoder support but someone beat us to it haha)

ReFil commented:
Hi guys is there anything more i need to do to get this merged? It's rebased to latest and has no conflicts so should be good to go

innovaker commented:
> Hi guys is there anything more i need to do to get this merged? It's rebased to latest and has no conflicts so should be good to go

Sorry for the delay and thank you for your patience.  We're in the midst of upgrading shield submissions so that we can accept shield PRs in parallel rather than series (which created the backlog you see today).  See #653 which is the leading edge of that overhaul.  I'm guessing this PR will be accepted after that's completed otherwise we're shooting a moving target.

In the meantime, this PR seems to replace all the line endings in  `docs/static/setup.ps1` - which it shouldn't.  You might want to check that.

ReFil commented:
oh god dammit not again. I had that issue last time i tried to submit a pr. I didn't even change anything in setup.ps1 haha

petejohanson reviewed (approved):
OH YEAH!
